File: Intent2/app/src/main/java/Modelo/Pizza.java
package Modelo;

import java.util.ArrayList;

public class Pizza {
    private String nombre;
    private ArrayList<String> ingredientes;
    private String tamaño;

    public Pizza(String nombre, String ingrediente1, String ingrediente2, String ingrediente3, String ingrediente4, String tamaño) {
        this.nombre = nombre;
        this.tamaño = tamaño;
        ingredientes = new ArrayList<>();
        ingredientes.add(ingrediente1);
        ingredientes.add(ingrediente2);
        ingredientes.add(ingrediente3);
        ingredientes.add(ingrediente4);
    }

    public ArrayList<String> getIngredientes() {
        return ingredientes;
    }

    public void setIngredientes(ArrayList<String> ingredientes) {
        this.ingredientes = ingredientes;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String obtenerStringIngredientes() {
        StringBuilder resp = new StringBuilder();
        int cont = 0;
        for (String ingrediente : ingredientes) {
            resp.append(cont == 1 ? ingrediente + "\n" : (cont == 3 ? ingrediente : ingrediente + ", "));
            cont++;
        }
        return resp.toString();
    }

    public String getTamaño() {
        return tamaño;
    }

    public void setTamaño(String tamaño) {
        this.tamaño = tamaño;
    }
}

File: Pizzeria/app/src/main/java/Modelo/Pizza.kt
package Modelo

class Pizza {

    private var nombre: String? = null
    private var ingredientes: ArrayList<String>? = null
    private var tamaño: String? = null


    constructor(
        nombre: String?,
        ingrediente1: String,
        ingrediente2: String,
        ingrediente3: String,
        ingrediente4: String,
        tamaño: String?
    ) {
        this.nombre = nombre
        this.tamaño = tamaño
        ingredientes = ArrayList()
        ingredientes!!.add(ingrediente1)
        ingredientes!!.add(ingrediente2)
        ingredientes!!.add(ingrediente3)
        ingredientes!!.add(ingrediente4)
    }

    fun getIngredientes(): ArrayList<String>? {
        return ingredientes
    }

    fun setIngredientes(ingredientes: ArrayList<String>?) {
        this.ingredientes = ingredientes
    }

    fun getNombre(): String? {
        return nombre
    }

    fun setNombre(nombre: String?) {
        this.nombre = nombre
    }

    fun obtenerStringIngredientes(): String? {
        var resp = ""
        var cont = 0
        for (ingrediente in ingredientes!!) {
            resp += if (cont == 1) {
                "$ingrediente\n"
            } else if (cont == 3) {
                "$ingrediente"
            } else {
                "$ingrediente, "
            }
            cont++
        }
        return resp
    }

    fun getTamaño(): String? {
        return tamaño
    }

    fun setTamaño(tamaño: String?) {
        this.tamaño = tamaño
    }
}